### ISSUE TITLE
Panic in v0.7

### ISSUE THREAD

**Issue by Boshen** [tamasfe/taplo#561]
npm version `@taplo/cli@v0.7.0`, in the Rspack repo:

> monorepo@0.5.4 format:toml /Users/bytedance/github/rspack
> npx @taplo/cli format '.cargo/*.toml' './crates/**/Cargo.toml' './Cargo.toml'

```
 INFO taplo:format_files:load_config: found configuration file path="/Users/bytedance/github/rspack/.taplo.toml"
 INFO taplo:format_files:collect_files: found files total=60 excluded=0 cwd="/Users/bytedance/github/rspack"
panicked at /Users/JounQin/Workspaces/GitHub/taplo/crates/taplo/src/formatter/mod.rs:684:25:
assertion failed: entry.comment.is_none()
```

https://github.com/tamasfe/taplo/blob/fe594e9cb1a1a1b0291e0dfb336b034870c7606b/crates/taplo/src/formatter/mod.rs#L684

It seems like you're building the wasm version on your own machine without the `--release` flag ;-)

Latest binary version v0.9.0 from `cargo` doesn't crash, so perhaps just a new release to npm should fix the problem.

**Comment by JounQin** [comment]
Duplicate of #559

**Comment by JounQin** [comment]
Blocked by https://github.com/tamasfe/taplo/pull/547#issuecomment-1950986876

**Comment by zphrs** [comment]
@JounQin I hit this error today in VSCode using the bundled repo and ended up switching to installing the cli via homebrew. Any updates on this?

**Comment by JounQin** [comment]
This is a very old issue, and vscode extension has already been published: https://marketplace.visualstudio.com/items/tamasfe.even-better-toml/changelog

**Comment by Nate711** [comment]
Getting this issue again with VS code insiders Version: 1.109.0-insider (Universal)
Commit: 1656700d1db06e50d22e733b890f2d29c23f70d9

version 0.21.2 of even better toml
```
panicked at /Users/JounQin/Workspaces/GitHub/taplo/crates/taplo/src/formatter/mod.rs:684:25:
assertion failed: entry.comment.is_none()

Stack:

Error
    at EA.G.wbg.__wbg_new_abda76e883ba8a5f (/Users/nathankau/.vscode-insiders/extensions/tamasfe.even-better-toml-0.21.2/dist/server.js:2:4187)
    at console_error_panic_hook::Error::new::hd83448d5e1a1b88a (wasm://wasm/065ed00a:wasm-function[40242]:0x1249bc2)
    at console_error_panic_hook::hook_impl::h24434d5d3a31a46a (wasm://wasm/065ed00a:wasm-function[10411]:0xcf17fe)
    at console_error_panic_hook::hook::hb562c91cafc2c895 (wasm://wasm/065ed00a:wasm-function[59765]:0x12e59c7)
    at core::ops::function::Fn::call::h46d6beedf89bee9d (wasm://wasm/065ed00a:wasm-function[42879]:0x1275769)
    at std::panicking::rust_panic_with_hook::h8e8556df5e46d2ff (wasm://wasm/065ed00a:wasm-function[20551]:0xfc8712)
    at std::panicking::begin_panic_handler::{{closure}}::h142b9f459ded72d9 (wasm://wasm/065ed00a:wasm-function[25954]:0x10c54b9)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h1c0d3278570375be (wasm://wasm/065ed00a:wasm-function[61550]:0x12e9f20)
    at rust_begin_unwind (wasm://wasm/065ed00a:wasm-function[42801]:0x12744d7)
    at core::panicking::panic_fmt::hbea4287e435ed8a3 (wasm://wasm/065ed00a:wasm-function[43645]:0x1280e86)


wasm://wasm/065ed00a:1


RuntimeError: unreachable
    at __rust_start_panic (wasm://wasm/065ed00a:wasm-function[63493]:0x12ec0f3)
    at rust_panic (wasm://wasm/065ed00a:wasm-function[56261]:0x12dbf09)
    at std::panicking::rust_panic_with_hook::h8e8556df5e46d2ff (wasm://wasm/065ed00a:wasm-function[20551]:0xfc8742)
    at std::panicking::begin_panic_handler::{{closure}}::h142b9f459ded72d9 (wasm://wasm/065ed00a:wasm-function[25954]:0x10c54b9)
    at std::sys_common::backtrace::__rust_end_short_backtrace::h1c0d3278570375be (wasm://wasm/065ed00a:wasm-function[61550]:0x12e9f20)
    at rust_begin_unwind (wasm://wasm/065ed00a:wasm-function[42801]:0x12744d7)
    at core::panicking::panic_fmt::hbea4287e435ed8a3 (wasm://wasm/065ed00a:wasm-function[43645]:0x1280e86)
    at core::panicking::panic::haa9f794b7d1c74c9 (wasm://wasm/065ed00a:wasm-function[41361]:0x125d8f9)
    at taplo::formatter::add_entries::hed6838edc1b01c33 (wasm://wasm/065ed00a:wasm-function[437]:0x3e5c50)
    at taplo::formatter::format_root::hcac06d9d442d0aa3 (wasm://wasm/065ed00a:wasm-function[256]:0x2eb282)

Node.js v22.21.1
[Error - 9:16:28 PM] Server process exited with code 1.
[Info  - 9:16:28 PM] Connection to server got closed. Server will restart.
true
 INFO registered request handler method="initialize"
 INFO registered request handler method="textDocument/foldingRange"
 INFO registered request handler method="textDocument/documentSymbol"
 INFO registered request handler method="textDocument/formatting"
 INFO registered request handler method="textDocument/completion"
 INFO registered request handler method="textDocument/hover"
 INFO registered request handler method="textDocument/documentLink"
 INFO registered request handler method="textDocument/semanticTokens/full"
 INFO registered request handler method="textDocument/prepareRename"
 INFO registered request handler method="textDocument/rename"
 INFO registered notification handler method="initialized"
 INFO registered notification handler method="textDocument/didOpen"
 INFO registered notification handler method="textDocument/didChange"
 INFO registered notification handler method="textDocument/didSave"
 INFO registered notification handler method="textDocument/didClose"
 INFO registered notification handler method="workspace/didChangeConfiguration"
 INFO registered notification handler method="workspace/didChangeWorkspaceFolders"
 INFO registered request handler method="taplo/convertToJson"
 INFO registered request handler method="taplo/convertToToml"
 INFO registered request handler method="taplo/listSchemas"
 INFO registered request handler method="taplo/associatedSchema"
 INFO registered notification handler method="taplo/associateSchema"
```

**Comment by peacefulotter** [comment]
@Nate711 same, I am having the exact same issue on latest VSCode and Even Better TOML
